### PR TITLE
Add ES6 Classes samples

### PR DESCRIPTION
**Pull request by addyosmani**
reviewers: @jeffposnick
Also welcome feedback from: @rauschma

Limitations noted in the intent-to-ship, which impact the scope of what was implemented here:

> - At this time we do not support subclassing built-ins and DOM objects (I've added a sample, but commented out)
> - Current ES6 specification (in particular, [[CreateAction]] hook that takes arguments) makes design of subclassable exotic (in particular DOM) objects unclear. We intend to work with involved parties on resolving this difficulty, but for now we add an extra limitation to calls to super constructor in subclasses: if a subclass constructor calls a superclass constructor via ‘super(...)’ call, that call must be the first statement of constructor’s body, and arguments to ‘super(...)’ call can not refer to ‘this’. 
>   We intend to lift this restriction in the future, once we clarify the design for exotic objects’ subclassing.

Note: these samples are intentional minimal to lower the friction involved in getting beginners on-board with the syntax.

### CODE REVIEW

File: classes-es6/index.html [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+        let inst = new MyPoly();
+        inst.getPolyName();
+
+
+        // Example 3: Extending an existing class
+        // ===============================================================
+
+        // Classes support extending other classes, but can also extend
+        // other objects. Let's extend the Polygon class to create a new
```
Comment (rauschma) [inline classes-es6/index.html]: Whatever you extend must be a constructor.

File: classes-es6/index.html [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        // derived class called Square.
+        class Square extends Polygon {
+            constructor(length) {
+                // The reserved 'super' keyword allows access to parent methods.
+                // Here, it will call the parent class' constructor with lengths
+                // provided for the Polygon's width and height
+                super(length, length);
```
Comment (rauschma) [inline classes-es6/index.html:147]: Note: `super()` for making super-constructor calls and `super.prop` for accessing super-properties and making super-method calls are quite different. But maybe that’s too subtle to be explained here.

Possibly: Mention what the default constructor for derived classes looks like.

**Comment by jeffposnick** [comment]
I'll definitely defer to @rauschma on the technical content. Nothing specific to add otherwise on my end. :+1:

**Comment by addyosmani** [comment]
Thanks for the reviews @rauschma and @jeffposnick. I'll get the comments addressed shortly.

Side: It looks like built-ins may in fact work, however the V8 unit tests appear to show limited support. I'll follow up with the team and update the samples here accordingly.

**Pull request merged**